src/rebar_file_utils.erl

```diff
@@ -273,6 +275,22 @@ canonical_path([_|Acc], [".."|Rest])  -> canonical_path(Acc, Rest);
 canonical_path([], [".."|Rest])       -> canonical_path([], Rest);
 canonical_path(Acc, [Component|Rest]) -> canonical_path([Component|Acc], Rest).
 
+%% returns canonical target of path if path is a link, otherwise returns path
+-spec resolve_link(string()) -> string().
+
+resolve_link(Path) ->
+    case file:read_link(Path) of
+        {ok, Target} ->
+            canonical_path(filename:absname(Target, filename:dirname(Path)));
+        {error, _} -> Path
+    end.
```

Comment:
Not sure :) I don't know how this project handles PRs like this - I'm
always inclined to get something small working first, then iterate. It's
pretty trivial to generalize this logic for use by other modules, but I'm
always inclined to take one step at a time.

On Tue, May 3, 2016 at 2:20 AM, James Fish <notifications@github.com> wrote:

> In src/rebar_file_utils.erl
> <https://github.com/erlang/rebar3/pull/1174#discussion_r61855986>:
>
> > @@ -273,6 +275,22 @@ canonical_path([_|Acc], [".."|Rest])  -> canonical_path(Acc, Rest);
> >  canonical_path([], [".."|Rest])       -> canonical_path([], Rest);
> >  canonical_path(Acc, [Component|Rest]) -> canonical_path([Component|Acc], Rest).
> >
> > +%% returns canonical target of path if path is a link, otherwise returns path
> > +-spec resolve_link(string()) -> string().
> > +
> > +resolve_link(Path) ->
> > +    case file:read_link(Path) of
> > +        {ok, Target} ->
> > +            canonical_path(filename:absname(Target, filename:dirname(Path)));
> > +        {error, _} -> Path
> > +    end.
>
> Re other providers, I don't know what a provider is :)
>
> Every task has an associate provider callback module that runs the task. I
> see this function is actually in a utility module so can be used by other
> providers. Other tasks that display source paths could benefit from this PR
> not just compile. Should these be configured globally or per task/provider?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/erlang/rebar3/pull/1174/files/da0e2c22959c976420b60def760b348aa200d6e2#r61855986>
>